WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingViewModel.kt

```diff
@@ -1,13 +1,60 @@
 package org.wordpress.android.ui.posts
 
+import android.os.Bundle
+import android.os.Parcelable
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import kotlinx.android.parcel.Parcelize
+import org.wordpress.android.ui.posts.ActionState.TagsActionState
+import org.wordpress.android.ui.posts.PrepublishingActionItemUiState.ActionType
+import org.wordpress.android.ui.posts.PrepublishingScreen.HOME
+import org.wordpress.android.ui.posts.PrepublishingScreen.TAGS
+import org.wordpress.android.viewmodel.Event
 import javax.inject.Inject
 
+const val KEY_TAGS_ACTION_STATE = "key_tags_action_state"
+
+enum class PrepublishingScreen {
+    HOME,
+    PUBLISH,
+    VISIBILITY,
+    TAGS
+}
+
+@Parcelize
+data class ActionsState(val currentScreen: PrepublishingScreen = HOME, val tagsActionState: TagsActionState) :
+        Parcelable
+
+sealed class ActionState : Parcelable {
+    @Parcelize
+    data class TagsActionState(val tags: String?) : ActionState()
+}
```

Comment:
I am utilizing these classes to represent all the state that can be persisted. My idea is that the current state will of each screen be saved if a configuration change occur and restored as such. So far, this model only covers the`tags` screen.  
Based on my exploration the changes made to tags, visibility etc. will be persisted with `EditPostRepository` so this setup is mainly to save the state when the user is in the process of setting these options before navigating back.